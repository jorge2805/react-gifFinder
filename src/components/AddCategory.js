import React, { useState } from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {
  
    const [inputValue, setinputValue] = useState("")

    const onChangeInput = (e) =>{
        setinputValue(e.target.value)
    }

    const onSubmitForm = (e) =>{
        e.preventDefault();
        
        if (inputValue.trim().length >= 2) {
            setCategories(categories => [inputValue, ...categories]);
            setinputValue("");
        }        
    }

    return (    
    <>
    <form onSubmit={onSubmitForm}>
        <input
            type="text"
            value={inputValue}
            onChange = {onChangeInput}
        />
    </form>
    </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}