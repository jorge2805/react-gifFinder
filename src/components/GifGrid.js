import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifts } from '../helpers/getGifts';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';



export const GifGrid = ({category}) => {

    const {data,loading} = useFetchGifs(category);

    return (
    <>
    <h3 className='animate__animated animate__bounce'>{category}</h3>
    <p className='animate__animated animate__flash'>{loading ? "Cargando..." : null}</p>
    <div className='card-grid'>       
        {
            data.map( image =>                
                <GifGridItem 
                key={image.id}
                {...image}                
                />
            )                
        }
    </div>
    </>
  )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
