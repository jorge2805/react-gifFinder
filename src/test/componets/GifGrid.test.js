import React from "react"
import { shallow } from "enzyme";

import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas en GifGrid', () => {
    
    
    test('Debe ser igual al SnapShot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category="Goku" /> );
        expect(wrapper).toMatchSnapshot();
    });    

    test('debe de generar items cuando se hace fetch ', () => {
        
        const gift =[{
            id: 'Hola',
            url: 'url',
            title: 'Titulo'
        }]

        useFetchGifs.mockReturnValue({
            data: gift,
            loading: false
        });
        const wrapper = shallow( <GifGrid category="Goku" /> );

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(true);
        expect(wrapper.find('GifGridItem').length).toBe(gift.length);
    });
})