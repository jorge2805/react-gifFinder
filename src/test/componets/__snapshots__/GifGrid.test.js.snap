// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe ser igual al SnapShot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en GifGrid debe de generar items cuando se hace fetch  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  />
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Hola"
      key="Hola"
      title="Titulo"
      url="url"
    />
  </div>
</Fragment>
`;
